### DISCUSSION TITLE
Speed?

### DISCUSSION THREAD

**Discussion by ACil32** [bertmelis/espMqttClient#167] [Q&A]
Ive heard this library was made with the idea of improving the [async-mqtt-client] library, but one of the things about that library was its speed. So my question is how is the speed in this library when dealing with large payloads, how does it compare to [async-mqtt-client?

**Comment by bertmelis** [comment]
I've never compared them. And it depends on the platform: async-mqtt-client was created in esp8266-times. There was no esp32 yet so there were no tasks, not task priorities...

warning: gut feeling ahead

- async-mqtt-client is faster on esp8266
- espMqttClient is faster for qos > 0
- asynctcp is faster than sync (wificlient libs)

Mind that espMqttClient supports payloads that don't fit it memory.

I once tried to do a test but the speed was going to link layer speed so I was testing WiFi instead of the library.

**Comment by bertmelis** [comment]
If speed and very large payloads are of utmost importancy, MQTT might not be the right protocol for your application.
